02-peer-protocol.md

```diff
@@ -952,22 +952,23 @@ Otherwise, on reconnection, a node MUST transmit `channel_reestablish`
 for each channel, and MUST wait for to receive the other node's
 `channel_reestablish` message before sending any other messages for
 that channel.  The sending node MUST set `commitments_received` to the
-number of `commitment_signed` messages received, and MUST set
-`revocations_received` to the number of `revoke_and_ack` messages
-received.
-
-If `commitments_received` is one less than the number of
-`commitment_signed` messages the receiving node has sent, it MUST send
-a set of `update_` messages against the prior commitment transaction
-followed by `commitment_signed`, otherwise if `commitments_received`
-is not equal to the number of `commitment_signed` messages the
-receiving node has sent, it SHOULD fail the channel.
-
-If `revocations_received` is one less than the number of
-`revoke_and_ack` messages the receiving node has sent, it MUST re-send
+commitment number of the last `commitment_signed` it received, and
+MUST set `revocations_received` to the commitment number of the last
+`revoke_and_ack` message received.
+
+If `commitments_received` is one less than the commitment number of
+the last `commitment_signed` message the receiving node has sent, it
+MUST send a set of `update_` messages against the prior commitment
+transaction followed by `commitment_signed`, otherwise if
+`commitments_received` is not equal to the commitment number of the
+last `commitment_signed` message the receiving node has sent, it
+SHOULD fail the channel.
+
```

Comment:
Always resend updates then commitment signed. And counters in reestablish msg make it clear whether it received previous or not.